### DISCUSSION TITLE
Disable/fix the taskbar overflow menu

### DISCUSSION THREAD

**Discussion by stamminator** [ramensoftware/windhawk-mods#1099] [Mod ideas]
The Windows 11 taskbar overflow menu is [very poorly designed](https://answers.microsoft.com/en-us/windows/forum/all/how-to-disable-taskbar-overflow-menu-and-restore/459e7adc-c85a-4e2a-8dc7-a14eadcdca32).

![zGazT48EO0](https://github.com/user-attachments/assets/f568a36c-dbc5-4227-9d17-afcf772758a9)

In this screenshot, the following issues can be observed:

- The overflow menu takes up more space than it's presumably intended to save
- My single window of XnView should not appear in the overflow menu at all since it's still visible on the taskbar, effectively making it look like there's a second window instance when there's just one.
- Items in the overflow menu do not honor the user's button/label combining settings.

Other issues I've observed include behaving differently on the primary monitor than it does on secondary monitors, plus the background occasionally being 100% transparent instead of acrylic.

Maybe a Windhawk mod could address this by allowing the user to either disable the taskbar overflow menu entirely, or allow them to apply the following fixes:

- Don't leave empty space to the right of the `...` button. It should always be right next to the tray hidden icons arrow.
- Don't show duplicate items. If an item is visible on the taskbar, then it shouldn't be in the overflow menu.
- Honor the user's "combine taskbar buttons and hide labels" setting.
- Only activate the overflow menu if it's actually going to free up more space than it takes up.

**Comment by m417z** [comment]
> Maybe a Windhawk mod could address this by allowing the user to either disable the taskbar overflow menu entirely

The Taskbar Labels mod had an option that made it appear only when the items become very small, but it stopped working after a recent update. Relevant issue: https://github.com/ramensoftware/windhawk-mods/issues/1097.

**Comment by moon-cacke** [comment]
use https://www.startallback.com/

**Comment by aetonsi** [comment]
very poorly designed is a HUGE understatement, this is CRAZY
![image](https://github.com/user-attachments/assets/86812785-0944-4af1-aef2-01c54c0a03af)

**Reply by stamminator** [reply]
God that is so bad

**Comment by jiucenglou** [comment]
Is there anyway to disable the overflow completely using windhawk ? I prefer win10 style when I can know (or guess) what are opened. When  windows are put in the overflow, I don't know if I did not open it or it is hidden.

**Reply by m417z** [reply]
Currently, there's no way to disable the taskbar overflow completely. There are the following ways that can help:
* The Taskbar Labels mod has an option that makes the taskbar overflow appear only when the items become very small.
* The Multirow Taskbar mod.

**Comment by aetonsi** [comment]
Apparently there's talks about it being actually fixed by M$ themselves? 

https://answers.microsoft.com/en-us/windows/forum/windows_11-start-win_taskbar/how-to-disable-taskbar-overflow-menu-and-restore/459e7adc-c85a-4e2a-8dc7-a14eadcdca32?messageId=97007587-424b-4239-9e74-78d31f8080ba
```
With today's update the Overflow Menu has gone!
It took 6 months of insults to have it removed.
A 30 minutes brainstorm would have been sufficient to not even implement it.

Time now to start a new crusade: remove the "suggested items" in start menu.
```

**Reply by burnedoutman** [reply]
they deleted the topic. does anyone remember/know the kb number where overflow was deleted?
win 11 23h2 22631.5624 still there

**Comment by albino1** [comment]
The thing that drives me crazy about the overlfow menu is that if I have a bunch of windows open on my 4K desktop, and then I remote it on my 1080p laptop or phone, those windows get dumped into the overflow, but then when I log back into my desktop they're still just stuck in overflow and won't come out.  Even if I start closing windows it will often just never come back out of overflow despite there being plenty of room on the taskbar.

**Comment by EleventhEric** [comment]
Like m417z mentioned in a previous comment, this can be fixed with the Taskbar Labels for Windows 11 mod by setting "Minimum taskbar item width" to something small. I set mine to 10 and now I can pretty much open as many programs as I want without seeing the Windows 11 overflow bug. It's the best option we have until Microsoft decides to fix this.
